File: experiments/source/python/book_imbalance.py
import loom


@loom.entry
def book_imbalance(epsilon: float, bid: loom.Tensor1, ask: loom.Tensor1) -> loom.Tensor1:
    return loom.Tensor1.map2(lambda bi, ai: (bi - ai) / (bi + ai + epsilon), bid, ask)

File: experiments/source/cpp/book_imbalance.cpp
#include <loom/loom.hpp>

LOOM_ENTRY loom::Tensor1 book_imbalance(float epsilon, loom::Tensor1 bid, loom::Tensor1 ask) {
  auto imbalance = [=](float bi, float ai) -> float {
    return (bi - ai) / (bi + ai + epsilon);
  };
  return loom::Tensor1::map2(imbalance, bid, ask);
}
